mrmustard/math/backend_tensorflow.py

```diff
@@ -87,6 +87,9 @@ def boolean_mask(self, tensor: tf.Tensor, mask: tf.Tensor) -> Tensor:
     def cast(self, array: tf.Tensor, dtype=None) -> tf.Tensor:
         if dtype is None:
             return array
+
+        if dtype not in [self.complex64, self.complex128, "complex64", "complex128"]:
+            array = self.real(array)
```

Comment:
what if we cast to int32?

Reply:
I only see warnings when I convert from complex to something else, and this case is covered.
Do you have a specific example in mind that is not covered?